sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/UtilsImpl.java

```diff
@@ -155,29 +152,35 @@ public static HttpPipeline buildHttpPipeline(
         HttpLoggingPolicy loggingPolicy = new HttpLoggingPolicy(logOptions);
         ArrayList<HttpPipelinePolicy> credentialPolicies = clone(policies);
         HttpPolicyProviders.addAfterRetryPolicies(credentialPolicies);
+        credentialPolicies.addAll(perRetryPolicies);
```

Comment:
Could we make separate methods for building the credential pipeline and the request pipeline. Having both together could lead to an easy mistake that breaks stuff.